hbase-server/src/main/java/org/apache/hadoop/hbase/coordination/ZKSplitTransactionCoordination.java

```diff
@@ -222,9 +229,21 @@ public void completeSplitTransaction(final RegionServerServices services, Region
     // Tell master about split by updating zk. If we fail, abort.
     if (coordinationManager.getServer() != null) {
       try {
-        zstd.setZnodeVersion(transitionSplittingNode(parent.getRegionInfo(), a.getRegionInfo(),
-          b.getRegionInfo(), coordinationManager.getServer().getServerName(), zstd,
-          RS_ZK_REGION_SPLITTING, RS_ZK_REGION_SPLIT));
+        int newNodeVersion = -1;
+        for (int i = 0; i < SPLIT_WAIT_TIMEOUT; i++) {
+          newNodeVersion = transitionSplittingNode(parent.getRegionInfo(), a.getRegionInfo(),
```

Comment:
Let me make an amendment to my earlier statement. I guess one way this could pass is if the participant we are connected to is behind the leader, and it catches up, and znode finally has our expectedVersion. But this could be accomplished with the sync() zk api before the read of the znode instead of a retry loop.

Reply:
To clarify reg this retry logic, I am just trying to add retry logic similar to above method `waitForSplitTransaction()` just to be consistent if we get BVE here while transitioning ZNode from RS_ZK_REGION_SPLITTING to RS_ZK_REGION_SPLIT, which we have not observed so far. The issue that we have observed has been with `waitForSplitTransaction()` but if there is a possibility of BVE here, we want to implement similar retry logic because if we fail to update ZNode state to RS_ZK_REGION_SPLIT here, we are going to get RIT for sure.

However, yes sync() ZK API calls sounds more reasonable

Reply:
Now that I revisited this, I agree that there is no point of this retry loop. Also, sync() is already done by `ZKAssign.transitionNode()`
```
    String node = getNodeName(zkw, encoded);
    zkw.sync(node);
```
Will remove this retry logic